DigitalLearningSolutions.Data/DataServices/UserDataService/UserCentreDetailsDataService.cs

```diff
@@ -15,7 +15,13 @@ public partial class UserDataService
             IDbTransaction? transaction = null
         )
         {
-            if (EmailIsInUseByOtherUser(userId, email))
+            if (transaction == null)
+            {
+                connection.Open();

```

Comment:
I think it'd be good to add an extension method `EnsureOpen()` which checks before calling Open() - it does sound like other people have had issues with calling Open() when the connection is already open, so I'd rather not rely on the implementation. And it wouldn't be much extra work to be careful.